oap-server/analyzer/agent-analyzer/src/test/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceLatencyThresholdsAndWatcherTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.analyzer.provider.trace;
+
+import java.util.Optional;
+import java.util.Set;
+import org.apache.skywalking.oap.server.analyzer.provider.AnalyzerModuleProvider;
+import org.apache.skywalking.oap.server.configuration.api.ConfigChangeWatcher;
+import org.apache.skywalking.oap.server.configuration.api.ConfigTable;
+import org.apache.skywalking.oap.server.configuration.api.ConfigWatcherRegister;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TraceLatencyThresholdsAndWatcherTest {
+    private AnalyzerModuleProvider provider;
+
+    @Before
+    public void init() {
+        provider = new AnalyzerModuleProvider();
+    }
+
+    @Test
+    public void testInit() {
+        TraceLatencyThresholdsAndWatcher traceLatencyThresholdsAndWatcher = new TraceLatencyThresholdsAndWatcher(provider);
+        Assert.assertEquals(traceLatencyThresholdsAndWatcher.getSlowTraceSegmentThreshold(), 500);
+        Assert.assertEquals(traceLatencyThresholdsAndWatcher.value(), "500");
+    }
+
+    @Test(timeout = 20000)
+    public void testDynamicUpdate() throws InterruptedException {
+        ConfigWatcherRegister register = new MockConfigWatcherRegister(3);
+
+        TraceLatencyThresholdsAndWatcher watcher = new TraceLatencyThresholdsAndWatcher(provider);
+        register.registerConfigChangeWatcher(watcher);
+        register.start();
+
+        while (watcher.getSlowTraceSegmentThreshold() == 10000) {
+            Thread.sleep(2000);
+        }
+        assertThat(watcher.getSlowTraceSegmentThreshold(), is(1000));
+        assertThat(provider.getModuleConfig().getSlowTraceSegmentThreshold(), is(500));
```

Comment:
why `is(500)`? Shouldn't it be `1000`?

Reply:
hello, `provider.getModuleConfig().getSlowTraceSegmentThreshold()` is module config value, watcher can't change it. watcher has it own value.